### DISCUSSION TITLE
[Bug] Package currently doesn't work when used with NextJS

### DISCUSSION THREAD

**Discussion by KTSnowy** [microsoft/monaco-editor#3701] [Q&A]
### Reproducible in vscode.dev or in VS Code Desktop?

- [X] Not reproducible in [vscode.dev](https://vscode.dev) or VS Code Desktop

### Reproducible in the monaco editor playground?

- [X] Not reproducible in [the monaco editor playground](https://microsoft.github.io/monaco-editor/playground.html)

### Actual Behavior

![Screenshot 2022-04-03 201145](https://user-images.githubusercontent.com/73310001/161453080-3d701387-31ab-4690-9408-3bacabd95420.png)

Package is currently broken when used with NextJS, there seems to be no way to fix without modifying the package's source code.

### Expected Behavior

Expected to be able to use the package with NextJS, specially since there's a ReactJS sample in the samples folder

### Additional Context

It seems that NextJS requires that css files be imported as ".module.css"

**Comment by chrison94** [comment]
I think the reason might also be related to: https://github.com/vercel/next.js/issues/10975

**Solution:**  
make use of next-transpile-modules inside your next.config.js.

As an example:

```
const config = {
  webpack: config => {
    config.plugins.push(
      new MonacoWebpackPlugin({
        languages: ['javascript'],
        filename: 'static/[name].worker.js',
      }),
    );
    return config;
  },
[…]
}


const withTM = require('next-transpile-modules')(['monaco-editor']);

module.exports = withTM(config);
```

**Comment by acao** [comment]
The above example doesn't work when adapted to the latest next@13

```js
const MonacoWebpackPlugin = require("monaco-editor-webpack-plugin");

/** @type {import('next').NextConfig} */
const nextConfig = {
  transpilePackages: ['monaco-editor'],
  webpack(config, options) {
    config.plugins.push(new MonacoWebpackPlugin({
      languages: ['json'],
      filename: 'static/[name].worker.js',
    }));
    return config;
  },
};

module.exports = nextConfig;
```

I get this error in the console/error boundary and next fails

```
setting getter-only property "Lexer"
```

worth noting - despite using the above config with `json` instead of `javascript`, only `static/editor.worker.ts` shows up. it appears this failure happens in the transpilation stage

**Reply by acao** [reply]
resolution here:

https://codesandbox.io/p/sandbox/monaco-graphql-nextjs-96lqjw

**Comment by vinoMamba** [comment]
@acao   have you already solved this problem ?

**Reply by acao** [reply]
@vinoMamba yes, sorry here is a working version here with monaco-graphql from a few months ago. just next.js and monaco-editor and the webpack plugin!

https://codesandbox.io/p/sandbox/monaco-graphql-nextjs-96lqjw

**Comment by rsodre** [comment]
There's a next.js working example here:

https://github.com/react-monaco-editor/react-monaco-editor/issues/271#issuecomment-986612363
